### ISSUE TITLE
Make `PermitSignature` compatible with more testing frameworks than just DSTest

### ISSUE THREAD

**Issue by PaulRBerg** [Uniswap/permit2#202]
## Problem

The `PermitSignature` contract inherits from Forge Std's `Test.sol`:

https://github.com/Uniswap/permit2/blob/bbbc92f895049ca45c4b25a450f9d3e907659284/test/utils/PermitSignature.sol#L4

But [DSTest](https://github.com/dapphub/ds-test) is a peer dependency of Forge Std, so DSTest is also a peer dependency of Permit2, and your `PermitSignature` utils cannot be used with any other assertions library like [PRBTest](https://github.com/PaulRBerg/prb-test) or [Vulcan](https://github.com/nomoixyz/vulcan).

## Solution

The only thing Permit2 needs from `Test` is the `vm` instance, so we can address this issue by removing the inheritance and declaring a private instance of `vm`, like this:

```diff
- import {Test} from "forge-std/Test.sol";
+ import {Vm} from "forge-std/Vm.sol";

- contract PermitSignature is Test {
+ contract PermitSignature {
+    Vm private constant vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
    // --- snip --- //
}
```

This also has to be done in [Permitter.t.sol](https://github.com/Uniswap/permit2/blob/bbbc92f895049ca45c4b25a450f9d3e907659284/test/actors/Permitter.sol).

Related issue in Forge Std: https://github.com/foundry-rs/forge-std/issues/353

**Comment by PaulRBerg** [comment]
It just dawned on me that another way to address this issue would be to turn `PermitSignature` into a library that has the `vm` as the first parameter. Then, Permit2 users would be able to use `PermitSignature` like this:

```solidity
contract MyTest is PRBTest {
    using PermitSignature for Vm;
}
```

**Comment by PaulRBerg** [comment]
Closed by #205.
